test/e2e/lua/dynamic_configuration.go

```diff
@@ -110,15 +109,22 @@ var _ = framework.IngressNginxDescribe("Dynamic Configuration", func() {
 			ingress, err := f.KubeClientSet.ExtensionsV1beta1().Ingresses(f.IngressController.Namespace).Get("foo.com", metav1.GetOptions{})
 			Expect(err).ToNot(HaveOccurred())
 
+			resp, _, errs := gorequest.New().
+				Get(fmt.Sprintf("%s?id=should_handle_annotation_changes", f.IngressController.HTTPURL)).
+				Set("Host", "foo.com").
+				End()
+			Expect(len(errs)).Should(Equal(0))
+			Expect(resp.StatusCode).Should(Equal(http.StatusOK))
+
 			ingress.ObjectMeta.Annotations["nginx.ingress.kubernetes.io/load-balance"] = "round_robin"
 			_, err = f.KubeClientSet.ExtensionsV1beta1().Ingresses(f.IngressController.Namespace).Update(ingress)
 			Expect(err).ToNot(HaveOccurred())
-
 			time.Sleep(5 * time.Second)
+
 			log, err := f.NginxLogs()
 			Expect(err).ToNot(HaveOccurred())
 			Expect(log).ToNot(BeEmpty())
-			index := strings.Index(log, fmt.Sprintf("reason: 'UPDATE' Ingress %s/foo.com", f.IngressController.Namespace))
+			index := strings.Index(log, "id=should_handle_annotation_changes")
```

Comment:
This is the actual fix. I noticed that marking the start of logs using `reason: 'UPDATE' Ingress %s/foo.com` is not enough because some logs appear before that even though those logs are result of that update.

So now we mark the start of the logs we are interested in using a request with unique query string to this test.